app/config/services/repositories.yml

```diff
@@ -0,0 +1,26 @@
+services:
+  ClientRepository:
+    class: AppBundle\Entity\Client
+    factory: ["@doctrine.orm.entity_manager", getRepository]
+    arguments:
+    - AppBundle\Entity\Client
+
+  AppBundle\v2\Repository\ClientRepository:
+    class: AppBundle\v2\Repository\ClientRepository
+    arguments:
+    - '@doctrine.orm.entity_manager'
+    - '@UserRepository'
+
+  UserRepository:
+    class: AppBundle\Entity\User
+    factory: ["@doctrine.orm.entity_manager", getRepository]
+    arguments:
+      - AppBundle\Entity\User
+
+  AppBundle\v2\Repository\DeputyRepository:
+    class: AppBundle\v2\Repository\DeputyRepository
+    arguments:
+      - '@doctrine.orm.entity_manager'
```

Comment:
If the Repository extends AbstractEntityRepository then it has the entity manager already injected via $this->_em

Reply:
Technically yes, but composition should be favoured over inheritance. Extending a class (which is already an extension) just to get at a property within a parent is a code smell and probably means we should inject the service we need.